packages/react/src/ActionList/LinkItem.tsx

```diff
@@ -21,53 +21,60 @@ type LinkProps = {
 }
 
 // LinkItem does not support selected, variants, etc.
-export type ActionListLinkItemProps = Pick<ActionListItemProps, 'active' | 'children' | 'sx' | 'inactiveText'> &
+export type ActionListLinkItemProps = Pick<
+  ActionListItemProps,
+  'active' | 'children' | 'sx' | 'inactiveText' | 'loading'
+> &
   LinkProps
 
-export const LinkItem = React.forwardRef(({sx = {}, active, inactiveText, as: Component, ...props}, forwardedRef) => {
-  const styles = {
-    // occupy full size of Item
-    paddingX: 2,
-    paddingY: '6px', // custom value off the scale
-    display: 'flex',
-    flexGrow: 1, // full width
-    borderRadius: 2,
+export const LinkItem = React.forwardRef(
+  ({sx = {}, active, inactiveText, loading, as: Component, ...props}, forwardedRef) => {
+    const styles = {
+      // occupy full size of Item
+      paddingX: 2,
+      paddingY: '6px', // custom value off the scale
+      display: 'flex',
+      flexGrow: 1, // full width
+      borderRadius: 2,
 
-    // inherit Item styles
-    color: 'inherit',
-    '&:hover': {color: 'inherit', textDecoration: 'none'},
-  }
+      // inherit Item styles
+      color: 'inherit',
+      '&:hover': {color: 'inherit', textDecoration: 'none'},
+    }
 
-  return (
-    <Item
-      active={active}
-      sx={{paddingY: 0, paddingX: 0}}
-      inactiveText={inactiveText}
-      data-inactive={inactiveText ? true : undefined}
-      _PrivateItemWrapper={({children, onClick, ...rest}) => {
-        const clickHandler = (event: React.MouseEvent<HTMLElement>) => {
-          onClick && onClick(event)
-          props.onClick && props.onClick(event as React.MouseEvent<HTMLAnchorElement>)
-        }
-        return inactiveText ? (
-          <Box sx={merge(styles, sx as SxProp)} {...rest}>
-            {children}
-          </Box>
-        ) : (
-          <Link
-            as={Component}
-            sx={merge(styles, sx as SxProp)}
-            {...rest}
-            {...props}
-            onClick={clickHandler}
-            ref={forwardedRef}
-          >
-            {children}
-          </Link>
-        )
-      }}
-    >
-      {props.children}
-    </Item>
-  )
-}) as PolymorphicForwardRefComponent<'a', ActionListLinkItemProps>
+    return (
+      <Item
+        active={active}
+        sx={{paddingY: 0, paddingX: 0}}
+        inactiveText={inactiveText}
+        loading={loading}
```

Comment:
Should links be allowed to have a loading state as they'd technically always be active? Are we expecting the link `href` to change during or after loading has completed?

Reply:
Good point! We've moved away from loading states for links, and instead we've been advocating for consumers to just hide loading links. I'm going to remove it.